src/cli/decrypt-webpub.ts

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env node
+
+import fs from 'fs';
+import path from 'path';
+import { ReadiumLink } from '../WebpubManifestTypes/ReadiumLink';
+import { WebpubManifest } from '../WebpubManifestTypes/WebpubManifest';
+import chalk from 'chalk';
+import sade from 'sade';
+import ora from 'ora';
+import type Decryptor from '@nypl-simplified-packages/axisnow-access-control-web';
+import { AxisNowEncryptionScheme } from '../constants';
+const pkg = require('../../package.json');
+
+const log = (info: string, arg?: string) =>
+  `${chalk.bold(info)}${arg ? chalk.blue(arg) : ''}`;
+
+sade('decrypt-webpub <manifest> <dest> <book_vault_uuid> <isbn>', true)
+  .version(pkg.version)
+  .describe('Fully decrypt a local webpub with the AxisNow decryptor.')
+  .example(
+    './outputs/axisnow.json outputs/dickens-axisnow/encrypted 9FD5E27H-73C8-4156-A623-FCDF66020BF9 9781682280652'
+  )
+  .action(decryptWebpub)
+  .parse(process.argv);
+
+export default async function decryptWebpub(
+  manifestPath: string,
+  destination: string,
+  book_vault_uuid: string,
+  isbn: string
+) {
+  console.log(
+    chalk.red(
+      `
+███╗   ██╗██╗   ██╗██████╗ ██╗     
+████╗  ██║╚██╗ ██╔╝██╔══██╗██║     
+██╔██╗ ██║ ╚████╔╝ ██████╔╝██║     
+██║╚██╗██║  ╚██╔╝  ██╔═══╝ ██║     
+██║ ╚████║   ██║   ██║     ███████╗
+╚═╝  ╚═══╝   ╚═╝   ╚═╝     ╚══════╝
+`,
+      chalk.red.bold`
+   Webpub Decryptor 
+`
+    )
+  );
+  const spinner = ora('Resolving decryptor package');
+  const manifestUrl = path.resolve(manifestPath);
+  const output = path.resolve(destination);
+  spinner.info(log('Reading manifest from: ', manifestUrl));
+  spinner.info(log('Destination folder: ', output));
+  spinner.info(log('book_vault_uuid: ', book_vault_uuid));
+  spinner.info(log('isbn: ', isbn));
+  let decryptor: Decryptor | undefined = undefined;
+  try {
+    const Decryptor =
+      require('@nypl-simplified-packages/axisnow-access-control-web').default;
+    decryptor = await Decryptor.createDecryptor({ book_vault_uuid, isbn });
+  } catch (e) {
+    throw new Error('AxisNow Decryptor package is not available.');
+  }
+  if (typeof decryptor === 'undefined') {
+    throw new Error('Could not instantiate decryptor');
+  }
+
+  spinner.start('Decrypting files...');
+
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const manifest: WebpubManifest = require(manifestUrl);
+
+  const resources =
+    manifest.resources?.map((res) =>
+      decryptLink(res, manifestUrl, decryptor!, output)
+    ) ?? [];
+
+  const readingOrder = manifest.readingOrder.map((chapt) =>
+    decryptLink(chapt, manifestUrl, decryptor!, output)
+  );
+
+  await Promise.all([...resources, ...readingOrder]);
+  spinner.succeed();
+
+  spinner.start('Writing new manifest.json');
+
+  manifest.readingOrder = manifest.readingOrder.map(removeEncryptionProperty);
+  manifest.resources = manifest.resources?.map(removeEncryptionProperty);
```

Comment:
Hypothetically there is a lot of overlap between the files in `readingOrder` and `resources`, which means this may be duplicating a fair amount of work? I'm not sure if that has any practical impact, it's very possible that it does not, but for large encrypted books (and with many calls to this service) that might add up.

Reply:
I was under the impression that items in the `readingOrder` wouldn't also appear in the `resources` normally. Basically `resources` being for things that are not part of the `readingOrder`... could be wrong those. Anyway this isn't meant to be a production endpoint or anything, just a util for development

Reply:
Maybe I'm mixing up my standards with something from OPF files, but I thought it was at least possible for things from `readingOrder` to show up in `resources`.

Regardless if this is just being used for development and testing then I don't think there are any issues with merging this!